### PR TITLE
Disable c10::import(ns)

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35714 Add temporary impl_UNBOXED syntax sugar for unboxed-only defs.
* #35706 Add DispatchKey impl overload; remove use of torch::dispatch
* #35476 Add unboxedCallRedispatch
* **#35398 Disable c10::import(ns)**

This disables namespaced c10::import which is broken with custom
mobile op builds.  This is to help prevent people from accidentally
breaking the custom mobile build in a mysterious way; if they use
the longform version it will work.  Fixing the analyzer is tracked
in https://github.com/pytorch/pytorch/issues/35397

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D20680519](https://our.internmc.facebook.com/intern/diff/D20680519)

### CODE REVIEW

File: torch/csrc/utils/python_dispatch.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ void initDispatchBindings(PyObject* module) {
     if (name.empty()) {
       return std::make_unique<c10::Module>(torch::import());
     } else {
-      return std::make_unique<c10::Module>(torch::import(name));
+      return std::make_unique<c10::Module>(c10::_import_DOES_NOT_WORK_WITH_MOBILE_CUSTOM_BUILD(name));
```
Comment (ljk53) [inline torch/csrc/utils/python_dispatch.cpp]: where is this used?

**Review by ljk53 — APPROVED** [review]
Thanks!

File: torch/csrc/utils/python_dispatch.cpp (hunk 1 above)
Comment (ezyang) [reply torch/csrc/utils/python_dispatch.cpp]: It's used from `test/test_dispatch.py`, just for testing only. I didn't feel like rewriting the testing code to interpolate in the namespaces.

**Review by smessmer — APPROVED** [review]

**Comment by zou3519** [comment]
This PR might have broken the XLA build, it started failing with the following:
```
Mar 31 05:58:01 torch_xla/csrc/aten_xla_type_default.cpp:9467:40: error: too many arguments to function call, expected 0, have 1
Mar 31 05:58:01   static auto dispatch = torch::import("aten")
Mar 31 05:58:01                          ~~~~~~~~~~~~~ ^~~~~~
Mar 31 05:58:01 /opt/conda/lib/python3.6/site-packages/torch/include/ATen/core/op_registration/op_registration.h:904:1: note: 'import' declared here
Mar 31 05:58:01 inline Module import() {
Mar 31 05:58:01 ^
```
cc @ailzhang

**Pull request closed without merging**